### ISSUE TITLE
Implement storage APIs for web driver

### ISSUE THREAD

**Issue by dweymouth** [fyne-io/fyne#4634]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

The storage APIs are currently unimplemented for web.

Related: #2734 

### Describe the solution you'd like to see.

Storage APIs implemented for web. Local storage will probably not be usable here due to size limitations.

**Comment by dweymouth** [comment]
This could probably be done with the [IndexedDB API](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) which is suitable for storing larger amounts of data than the 5MB-10MB limit of localStorage

**Comment by dweymouth** [comment]
This [comment](https://github.com/fyne-io/fyne/issues/2734#issuecomment-1939732591) from @kkoreilly may be relevant here

**Comment by dweymouth** [comment]
It's probably more heavyweight (eg we don't need unix permissions, etc) and pulls in more dependencies than we want, but could be a good guide for how to build our own filesystem-like abstraction on indexeddb

**Comment by kkoreilly** [comment]
Yep, if you're not trying to implement the stdlib os API, you should need a lot less code. Most of the logic in jsfs is implementing different os functions that aren't needed for simple file storage.

**Comment by bilalshaikh292** [comment]
Hey, I am new to this Org . I just wanted to know that is this issue open and if anyone is already working on this.

**Comment by andydotxyz** [comment]
Nobody was working on this - however it is part of the upcoming 2.5 release which is due imminently. So we are looking for a quick turnaround (i.e. days remaining) - is that possible?

**Comment by bilalshaikh292** [comment]
@andydotxyz just got a need to work on a university Project. Sorry won't be able to do this. Thanks for support.

**Comment by andydotxyz** [comment]
Let's not push it that far. No new APIs required so we should be able to pull it into 2.5.x (hopefully the .1 at the latest, but who knows...)

**Comment by sdassow** [comment]
I'm working on this now.

**Comment by andydotxyz** [comment]
Thanks for this @sdassow - we will get the bounty paid out after release.
